### PR TITLE
Delete some unused cmake defines

### PR DESCRIPTION
**Pull request by am11**
Found with:

```sh
git grep cmakedefine ':/*.in' | while read x y z; do echo $y; done | sort -u |\
    while read x; do if ! git grep -q "$x" ':!*in' ':!*cmake'; then echo "$x";fi;done
```

### CODE REVIEW

File: src/mono/cmake/configure.cmake [hunk 1]
```diff
@@ -234,12 +234,9 @@ if(HOST_WIN32)
   # Dynamic lookup using ac_check_headers/ac_check_functions is extremly slow on Windows, espacially on msbuild.
   # Since majority of the checks above will fail on Windows host, we can just directly define the available static
   # API surface.
-  set(HAVE_ATEXIT 1)
```
Comment (steveisok) [inline src/mono/cmake/configure.cmake:237]: I think we need this and perhaps the others. Is that correct, @lateralusX ?

Comment (am11) [reply src/mono/cmake/configure.cmake:237]: These are unused. When we need them lets add them then.

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

File: src/mono/cmake/configure.cmake (hunk 1 above)
Comment (steveisok) [reply src/mono/cmake/configure.cmake:237]: In dotnet/runtime perhaps, but they may be used in forks. If that's the case, I'd rather keep it.

Comment (am11) [reply src/mono/cmake/configure.cmake:237]: Can you point me to the fork you are referring to? The dead code is deleted in this repo. If fork is using it, it should be kept there.

Comment (steveisok) [reply src/mono/cmake/configure.cmake:237]: It's non-public, unfortunately. I want to make sure it's even used there first. If it's not, then my comment is moot.

Comment (lateralusX) [reply src/mono/cmake/configure.cmake:237]: Not used so should be fine dropping and if needed in future we add it downstream.

File: src/mono/cmake/config.h.in [hunk 2]
```diff
@@ -630,9 +585,6 @@
 /* Host is RISC-V */
 #cmakedefine HOST_RISCV 1
 
-/* Host is 32-bit RISC-V */
-#cmakedefine HOST_RISCV32 1
```
Comment (akoeplinger) [inline src/mono/cmake/config.h.in:634]: I'd keep this one to have parity with TARGET_RISCV32. we always have both HOST_ and TARGET_ defines for architectures

**Review by akoeplinger — APPROVED** [review]

File: src/mono/cmake/config.h.in [hunk 3]
```diff
@@ -630,9 +585,6 @@
 /* Host is RISC-V */
 #cmakedefine HOST_RISCV 1
```
Comment (am11) [inline src/mono/cmake/config.h.in:587]:
```suggestion

/* Host is 32-bit RISC-V */
#cmakedefine HOST_RISCV32 1

```

**Pull request merged**